### DISCUSSION TITLE
Logging with Serilog

### DISCUSSION THREAD

**Discussion by justSteve** [oqtane/oqtane.framework#941] [Ideas]
I'm brand new to oqtane and no background in DNN either so I'm not versed in how opinions on the usage of 3rd party dependencies are approached. The logging in current distro currently has better than average logging but .net 5 and EF core versions bring new levels of instrumentation that dovetails really nicely with the structure afforded by using Serilog from the ground up. 

Are there any other souls out there that have toyed with Serilog  inside the oqtane base (or any other logging framework for that matter)?

**Comment by sbwalker** [comment]
Third party dependency usage is outlined in the Oqtane Philosophy: https://www.oqtane.org/Resources/Blog/PostId/538/oqtane-philosophy. Developers are free to use whatever third party dependencies they desire in their own custom modules and themes.

I looked at Serilog during the Oqtane POC phase. It worked great with a basic application but I was unable to get it to work properly with the multi-tenant features of Oqtane ( ie. the ability to have shared or isolated tenants ). It also did not have support for Blazor WebAssembly at that time - which was a critical requirement for Oqtane. And I also wanted a fully integrated and secure dashboard where a developer/admin could easily view the log events. The logging solution in Oqtane uses the Serilog concepts for structured logging but also supports the other items I outlined above.

**Reply by justSteve** [reply]
Understood. And the published 'philosophy' doc inspires confidence. Especially the focus on performance. thx.
